### DISCUSSION TITLE
Representing composition-based structure as nested

### DISCUSSION THREAD

**Discussion by thcrt** [mkdocstrings/mkdocstrings#721] [Q&A]
I'm trying to set up mkdocstrings to automatically generate docs for my project's config file. It generates docs based on my code that uses Pydantic to define the structure of the config file, looking something like this (with docstrings added, of course):

```python
class AuthSettings(BaseModel):
    authfoo: str
    authbar: int = 42

class Settings(BaseSettings):
    foo: str
    bar: int = 17
    
    auth: AuthSettings = AuthSettings()
```

So, `Settings` has a field `auth` which should refer to an instance of `AuthSettings`. 

Right now, all I can get working is something like this:

![image](https://github.com/user-attachments/assets/18dedd68-ff74-4b8c-af62-a358685f028d)

I don't want `AuthSettings` to be displayed at all -- it's an internal representation that doesn't mean anything to users. The fields currently documented underneath it should instead be nested under the `auth` field of `Settings`.

How can I generate docs that display this nested relationship properly?

**Comment by pawamoy** [comment]
That's a good question!

You could probably write a small Griffe extension that re-attaches members of `AuthSettings` as members of `Settings.auth`, however I think mkdocstrings-python templates won't display them (we didn't plan for displaying attributes of attributes), so you'd have to overwrite them to add the members rendering part:

```django
{% extends "_base/attribute.html.jinja" %}

{% block contents scoped %}
  {{ super() }}
  
  {% block children scoped %}
    {% set root = False %}
    {% set heading_level = heading_level + 1 %}
    {% include "children"|get_template with context %}
  {% endblock children %}
{% endblock contents %}
```
